### ISSUE TITLE
(Feature request) Other alternative databases as sources and the inclusion of PSP homebrew.

### ISSUE THREAD

**Issue by nowhereman87** [Ibrahim778/BetterHomebrewBrowser#38]
This app is one of the best options to get homebrew on our consoles and in my opinion it is on par with VitaDB downloader, since although it has more options, I like the BHBB interface most: the touch capability, it is visually more pleasant and easier to read, and especially because of the possibility of downloading in the background.

Another of the features that seems most notable to me is the ability to use both VitaDB and CBPSdb as sources to obtain VPKs and their data files, since as is known, not all the content is found in the different databases.

Would it be possible for you to incorporate other databases and PSP content, (VitaDB app list just a few) which is massive and forms a big part of the PSVita homebrew story?

I think it's important that a lot of that stuff is still accessible and preserved, plus the PSP scene never ends, and after all, those creations are natively usable on PSVita.

To achieve that I wanted to suggest if you could incorporate brewology.com as another alternative database, which although it seems to have been abandoned recently, has the largest amount of stuff of all kinds, especially from the early eras of both PSP and PSVita scenes, content that is very difficult to find anywhere else.

PSP: [](https://www.brewology.com/downloads/?dcid=1)
PSVita: [](https://store.brewology.com/vita/homebrew.php?lang=)

On the other hand, it would be good to also link to the content of [Gamebrew.org](https://www.gamebrew.org/wiki/Main_Page), which is a relatively new site but very well curated and very comprehensive, in addition to being made as a Wiki.

PSP: [](https://www.gamebrew.org/wiki/List_of_all_PSP_homebrew)
PSVita: [](https://www.gamebrew.org/wiki/List_of_all_Vita_homebrew)

Both sites have a lot of unique material that is not found in VitaDB or CBPS, but beyond that, I think it is good to have more sources as alternatives, which function as mirrors in the event of a crash or catastrophe in the common sources.

Thank you very much for all the work done so far!

**Comment by Ibrahim778** [comment]
Brewology was actually on my radar for quite some time, however I never got round to adding it because I simply couldn't find a convenient way to do so, as far as I'm aware it doesn't expose any readily available API for me to use
Gamebrew is quite impressive however, it also packs it's downloads differently meaning there'd be quite a bit more work to implement and it, too, doesn't expose any API's putting it in the same position as Brewology.
However, I may be able to add PSP homebrew support, seeing how PSP Homebrew Browser exists, but I cant seem to find any source code on how it's made so it's going to take some work

And I will say this, don't expect any updates in the foreseeable future, I have effectively dropped development on all my projects since I am now lacking the time to work on them.

**Comment by nowhereman87** [comment]
Hello, thank you for taking this into consideration, if you ever find a moment, could you try any of these repositories/portals as sources to be able to install PSP homebrew on PSvita straight from your app?

This mini-portal is a gem and very important because it is currently usable from the PSP itself, the homebrew is well organized by categories:
>> http://psp.lusidgames.com/
>> ( https://github.com/lusid1/psp )
>> http://brew.psp.place/hb/

This Github repository is the "official one" curated by the PSP homebrew community on Discord (https://discord.gg/VZcjFvy3mT), which today maintains some of the biggest developments such as CFW ARK and Nazi Zombies Portable etc:

>> https://github.com/PSP-Archive

This enormous collection is also maintained by its members in the Internet Archive must be the largest available along with Wololo's, the good thing is that being on this site helps the material be preserved in the long-term future:
>> https://archive.org/details/psp-homebrew-library?sort=-date

Just in case, another of the historical sites, PDRoms.de, now abandoned but with a huge archive still available:
>> https://pdroms.de/news/sony-playstationportable-psp/

Here a great list of other homebrew repositories and sites from the golden years: 
>> https://github.com/PSP-Archive/docs/blob/main/homebrew-lib/homebrew-sources.md

If you think that implementing any of this would overload the original app, perhaps you could consider doing a separate client for just the PSP content like the old Arkanite app was ( https://www.psx-place.com/resources/psp-homebrew-browser.580/ )

I hope you find the time to even make a little part of this historic stuff downloadable from Vita, greetings!

**Comment by Ibrahim778** [comment]
I have actually figured out the source for PSP homebrew browser and adding it to bhbb should very much be possible, however it isn't as extensive as I remember.
I have also found the psp-homebrew-library on archive.org to be a very viable source as well
However since both of these use new formats I have not yet added parsers for in BHBB (yml & xml) they will both take a considerable amount of effort to integrate nicely.

I had previously decided to take a step back from PS Vita development. However, I do have some more free time coming up if I decide to change my mind and come back, this will be the first thing I try out.

**Comment by nowhereman87** [comment]
Wow, thank you very much for the response and for making it happen so quickly, this is very important for this old material to remain in circulation, and thus to be preserved. I understand that working with larger and more complete sources like the Internet Archive must be more complicated, I hope one day it will be possible.

I was testing and it works perfectly, this is definetively the best option we have on our Vitas to download homebrew, the integration to the UI and the download in the background are perfect and give it an official quality to its use.
I don't want to abuse your generosity but could you perhaps map the select or square button to toggle between the different sources, or even to go directly to the PSP downloads? 

Thank you. I will do my best to spread the word abuout this app , I think it deserves much more recognition.

**Comment by Ibrahim778** [comment]
Thank you for your kind words.
I could map the select button to directly jump to the source selector but I don't really see a point. I'd like to remind you that the triangle button is already mapped to Triangle (and usually is in any PAF app) so is another keybinding really necessary to skip 1 more button press? 
I don't really like the idea of mapping a button to go directly to PSP homebrew.

**Comment by nowhereman87** [comment]
Yes I learned the triangle map, its cool. I know it sounds lazy, but my idea is to pass from one source to another, rotating, with a single button, like VitaDB downloader app does with L1. Thanks!

**Comment by nowhereman87** [comment]
> This mini-portal is a gem and very important because it is currently usable from the PSP itself, the homebrew is well organized by categories:
> 
> http://psp.lusidgames.com/
> ( https://github.com/lusid1/psp )
> http://brew.psp.place/hb/

Hi mate just wanted to mention that someone pointed to me on the main PSP discord server that this mini portal/store is In fact just an index to the main and official repository (the archive.org one) and so you could "just invoke the API" to include that massive PSP library in BHBB. I'm not saying it's simple but I thought it was important to mention it.

**Comment by devingDev** [comment]
> > This mini-portal is a gem and very important because it is currently usable from the PSP itself, the homebrew is well organized by categories:
> > http://psp.lusidgames.com/
> > ( https://github.com/lusid1/psp )
> > http://brew.psp.place/hb/
> 
> Hi mate just wanted to mention that someone pointed to me on the main PSP discord server that this mini portal/store is In fact just an index to the main and official repository (the archive.org one) and so you could "just invoke the API" to include that massive PSP library in BHBB. I'm not saying it's simple but I thought it was important to mention it.

What would be the exact priority list of all these sites, listed in this issue, to add?

**Comment by nowhereman87** [comment]
If you wanted to include more than just games, emulators and ports, it would be everything included in this index: https://brew.psp.place/hb/index.html
Although it seems that the base repository is one of the others I mentioned above (Github or the Internet Archive)

> This Github repository is the "official one" curated by the PSP homebrew community on Discord (https://discord.gg/VZcjFvy3mT), which today maintains some of the biggest developments such as CFW ARK and Nazi Zombies Portable etc:
> 
> > > https://github.com/PSP-Archive
> 
> This enormous collection is also maintained by its members in the Internet Archive must be the largest available along with Wololo's, the good thing is that being on this site helps the material be preserved in the long-term future:
> 
> > > https://archive.org/details/psp-homebrew-library?sort=-date
> 

Luckily the Internet Archive is already working again and it is also very complete.

Sorry but I am not a dev to clearly understand which structure would be more feasible and better to use, I think if any of those could be added it would be an big achievement since at the moment there is nothing like that for Vita. Thanks for the interest.
